_posts/2022-03-17-ols-4-participant-nadine-spychala.md

```diff
@@ -0,0 +1,113 @@
+
+OLS-4 has also got me started on open science project essentials like code of conduct and contributor guidance specifications.
+
+## The main goals achieved in this project
+What I’ve done:
+
+- developed software which allows applying one or more measures to one or multiple datasets in one line of code, and which delivers a pandas dataframe storing all results alongside respective parameters,
+
+```
+
+import complex_py as cp
+
+# define what data to use/generate, and what measures of emergence 
+# to apply to those data via the following five variables: 
+
+#       model_function:      contains functions to generate/load your data
+#       model_variables:     contains parameters for generating/loading your data
+#       emergence_functions: contains functions to calculate measures of emergence
+#       measure_variables:   contains parameters for calculating emergence
+#       parameters:          contains all parameters (data- and measure-related ones) 
+
+# compute measures of emergence for all parameter combination:
+emergence_df = cp.compute_emergence(model_functions, model_variables, 
+                                    emergence_functions, measure_variables, parameters)
+
+``` 
+
+- implemented one measure of emergence easily to be complemented by other measures as the code is scalable, 
+
+- started to implement code testing as well as visualization techniques of the results,   
+
+- started to think about the “format” in which to offer this library to potential users,
+
+- started to add/work on open science project essentials like a code of conduct or readme files.
+
+Still to do:
+
+- develop tutorials about a) how to use the library, including use cases, caveats, and anything that is important to keep in mind when applying the measures, and b) the measures themselves, including definitions, commonalities and differences that will be showcased in example models,
+
+- make a decision about how the library shall be used (e. g., directly in Python, via a command line interface or a Docker image),   
+
+- continue to work on visualizing results, code testing, as well as code of conduct and contributing guidelines (will be high priority once a first version is released which will     then be open to potential contributors),
+
+- implement further measures.
+
+## The initial steps
+At the point where OLS-4 started, I was still struggling to make a measure of emergence work in Python via oct2py() - a Python library which allows the calling of Matlab functions in Python. The battle continued for about 3 months after which – with a few hacky interventions – I was able to use it in Python. Clearly, a long-term solution will involve porting those Matlab functions into Python (something everyone my mentor and other code reviewers were heavily stressing)… but as this will be quite some task, it’s good to have this work-around in the meantime to progress on other parts of the project.
```

Comment:
```suggestion
At the point where OLS-4 started, I was still struggling to make a measure of emergence work in Python via `oct2py()`- a Python library which allows the calling of Matlab functions in Python. The battle continued for about 3 months after which – with a few hacky interventions – I was able to use it in Python. Clearly, a long-term solution will involve porting those Matlab functions into Python (something everyone my mentor and other code reviewers were heavily stressing)… but as this will be quite some task, it’s good to have this work-around in the meantime to progress on other parts of the project.
```